payment/src/main/java/org/killbill/billing/payment/invoice/InvoicePaymentControlPluginApi.java

```diff
@@ -129,7 +129,9 @@ public PriorPaymentControlResult priorCall(final PaymentControlContext paymentCo
         final InternalCallContext internalContext = internalCallContextFactory.createInternalCallContext(paymentControlContext.getAccountId(), paymentControlContext);
         switch (transactionType) {
             case PURCHASE:
-                return getPluginPurchaseResult(paymentControlContext, pluginProperties, internalContext);
+                // if the Payment Method is still null, abort payment
+                return (paymentControlContext.getPaymentMethodId() != null) ? getPluginPurchaseResult(paymentControlContext, pluginProperties, internalContext) :
```

Comment:
> Wouldn't we be automatically in `AUTO_PAY_OFF` today (unless maybe the account never had a payment method)?

It's indeed the only use-case today.

@javier-gomez-hs Can you add a `log.warn` though, to make sure administrators understand from the logs why the payment wasn't triggered (maybe print the `invoiceId` too)?